### ISSUE TITLE
IslandStatusChangeEvent issue

### ISSUE THREAD

**Issue by AhmHkn0** [craftaro/FabledSkyBlock#27]
When I register the class where I listen for the "IslandStatusChangeEvent" event, this error appears in the console
```
org.bukkit.plugin.IllegalPluginAccessException: java.lang.NullPointerException
        at org.bukkit.plugin.SimplePluginManager.getEventListeners(SimplePluginManager.java:705) ~[patched_1.16.5.jar:git-Purpur-1171]
        at org.bukkit.plugin.SimplePluginManager.registerEvents(SimplePluginManager.java:657) ~[patched_1.16.5.jar:git-Purpur-1171]
        at pw.reborncraft.rssynccore.main.registerEvents(main.java:37) ~[?:?]
        at pw.reborncraft.rssynccore.main.onEnable(main.java:27) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.5.jar:git-Purpur-1171]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.16.5.jar:git-Purpur-1171]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.16.5.jar:git-Purpur-1171]
        at com.rylinaux.plugman.util.BukkitPluginUtil.load(BukkitPluginUtil.java:428) ~[?:?]
        at com.rylinaux.plugman.util.BukkitPluginUtil.load(BukkitPluginUtil.java:385) ~[?:?]
        at com.rylinaux.plugman.util.BukkitPluginUtil.reload(BukkitPluginUtil.java:500) ~[?:?]
        at com.rylinaux.plugman.command.ReloadCommand.execute(ReloadCommand.java:125) ~[?:?]
        at com.rylinaux.plugman.PlugManCommandHandler.onCommand(PlugManCommandHandler.java:97) ~[?:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[patched_1.16.5.jar:git-Purpur-1171]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:172) ~[patched_1.16.5.jar:git-Purpur-1171]
        at org.bukkit.craftbukkit.v1_16_R3.CraftServer.dispatchCommand(CraftServer.java:826) ~[patched_1.16.5.jar:git-Purpur-1171]
        at org.bukkit.craftbukkit.v1_16_R3.CraftServer.dispatchServerCommand(CraftServer.java:788) ~[patched_1.16.5.jar:git-Purpur-1171]
        at net.minecraft.server.v1_16_R3.DedicatedServer.handleCommandQueue(DedicatedServer.java:483) ~[patched_1.16.5.jar:git-Purpur-1171]
        at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:450) ~[patched_1.16.5.jar:git-Purpur-1171]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1416) ~[patched_1.16.5.jar:git-Purpur-1171]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1142) ~[patched_1.16.5.jar:git-Purpur-1171]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:293) ~[patched_1.16.5.jar:git-Purpur-1171]
        at java.lang.Thread.run(Thread.java:833) [?:?]
[08:58:01 INFO]: [RSSyncCore] Disabling RSSyncCore v1.0
```

**Comment by SpraxDev** [comment]
I don't see any of our plugins/code in the stack trace you posted, so only seeing that it looks like a bug in your plugin.

I'd need to see the problematic code to be able to help in any way (for example main.java line 37).

**Comment by AhmHkn0** [comment]
How can there be a bug in my plugin:
```
 public void onEnable() {
        plugin = this;
        if (!getServer().getPluginManager().isPluginEnabled("FabledSkyBlock")) {
            getLogger().severe("FabledSkyBlock did not found. Plugin will not be enable.");
            return;
        } else
            getLogger().info("FabledSkyBlock founded. Enabling sync systems.");
        getServer().getPluginManager().registerEvents(new IslandEvents(), plugin);
}
```

```
public class IslandEvents implements Listener {

    @EventHandler
    public void islandStatusChange(IslandStatusChangeEvent event) {

    }

}
```

It works when I put another event on listening. There is only a problem with the StatusChange event.

**Comment by SpraxDev** [comment]
Ah I think I see what is causing the `IllegalPluginAccessException` and I commited a fix.

I've just released v3.0.4 of the plugin - Can you please update and try again?

**Comment by AhmHkn0** [comment]
It worked. Thanks.
